File: packages/ogdf.js/src/module/config/HierarchyClusterLayout.js
import createModule from '../module-helper'
import { PARAMETER_TYPE } from '../../utils/parameter-type'
const NAME = 'HierarchyClusterLayoutModule'

const MODULE_DIRECTORY = {
    OptimalHierarchyClusterLayout: {
        fixedLayerDistance: {
            type: PARAMETER_TYPE.BOOL,
            range: [true, false],
            default: false
        },
        layerDistance: {
            type: PARAMETER_TYPE.DOUBLE,
            range: [0, Infinity],
            default: 3.0
        },
        nodeDistance: {
            type: PARAMETER_TYPE.DOUBLE,
            range: [0, Infinity],
            default: 3.0
        },
        weightBalancing: {
            type: PARAMETER_TYPE.DOUBLE,
            range: [0, Infinity],
            default: 0.1
        },
        weightClusters: {
            type: PARAMETER_TYPE.DOUBLE,
            range: [0, Infinity],
            default: 0.05
        },
        weightSegments: {
            type: PARAMETER_TYPE.DOUBLE,
            range: [0, Infinity],
            default: 2.0
        }
    }
}

const HierarchyClusterLayoutModule = createModule(NAME, MODULE_DIRECTORY)
export default HierarchyClusterLayoutModule

File: packages/ogdf.js/src/module/src/HierarchyClusterLayout.cpp
#include "../main.h"

#include <ogdf/layered/OptimalHierarchyClusterLayout.h>

EM_PORT_API(OptimalHierarchyClusterLayout *)
HierarchyClusterLayoutModule_OptimalHierarchyClusterLayout(bool fixedLayerDistance, double layerDistance, double nodeDistance, double weightBalancing, double weightClusters, double weightSegments)
{
    OptimalHierarchyClusterLayout *optimalHierarchyClusterLayout = new OptimalHierarchyClusterLayout();
    optimalHierarchyClusterLayout->fixedLayerDistance(fixedLayerDistance);
    optimalHierarchyClusterLayout->layerDistance(layerDistance);
    optimalHierarchyClusterLayout->nodeDistance(nodeDistance);
    optimalHierarchyClusterLayout->weightBalancing(weightBalancing);
    optimalHierarchyClusterLayout->weightClusters(weightClusters);
    optimalHierarchyClusterLayout->weightSegments(weightSegments);
    return optimalHierarchyClusterLayout;
}
